### PR TITLE
addon repositories: fix last checked time not updating if not previously added

### PR DESCRIPTION
**Pull request by tamland**
As reported in http://forum.kodi.tv/showthread.php?tid=231092&pid=2111696#pid2111696

cc @popcornmix

### CODE REVIEW

**Comment by mkortstiege** [comment]
I can confirm that this fixes .zip installs. Thanks.

**Comment by tamland** [comment]
@mkortstiege its two separate issues (see #8111). Im pretty sure this has nothing to do with zip install problem

**Comment by mkortstiege** [comment]
@tamland you're right. I had an interim fix in my tree that fooled me.

**Comment by MilhouseVH** [comment]
@tamland: Thanks for this, but what exactly is this fixing based on the two problems reported in the forum post - is it the dead repository log spam, or the lists that update every 13-14 seconds? Or maybe it fixes both issues (that would be nice)?

**Comment by tamland** [comment]
@MilhouseVH  The problem here is that the `lastcheck` field in the database wasn't updating. In #7874 it was changed to store the time of any check, not just successful ones, but SQL only does a `UPDATE` not `INSERT` (for repos that are working or have successfully updated once in the past, this was never an issue as they were properly inserted via `AddRepository`).

When the updater checks for when to do the next update, it sees the broken repository as never updated and update again immediately, hence the infinite loop. 

The list updating is because of https://github.com/xbmc/xbmc/blob/382696a54cac9a07e312040f88a5dff6a9eb3697/xbmc/addons/RepositoryUpdater.cpp#L85 which is called after every update to propagate the changes. This is old code, and used a lot of place. I don't know if there are any better ways to do it.

tldr; it should fix both.

**Comment by MilhouseVH** [comment]
Excellent, many thanks - will include this in the next build.

**Comment by uNiversaI** [comment]
Will this also fix addon updates kay toast popping up while watching video?

**Comment by MartijnKaijser** [comment]
jenkins build this please
is this ready to go in?

**Comment by mkortstiege** [comment]
@MilhouseVH do you got some feedback for this?

**Comment by MilhouseVH** [comment]
> @MilhouseVH do you got some feedback for this?

No feedback so far (which probably means it's now fixed) - I've asked for an update/confirmation from someone that had the problem.

**Comment by MilhouseVH** [comment]
@mkortstiege: Yep, all good now - many thanks!
